src/main/java/com/teragrep/pth_06/planner/walker/conditions/IndexStatementCondition.java

```diff
@@ -0,0 +1,158 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://github.com/teragrep/teragrep/blob/main/LICENSE>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+
+package com.teragrep.pth_06.planner.walker.conditions;
+
+import com.teragrep.blf_01.Token;
+import com.teragrep.pth_06.config.ConditionConfig;
+import com.teragrep.pth_06.planner.StreamDBClient;
+import org.apache.spark.util.sketch.BloomFilter;
+import org.jooq.Condition;
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Element;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+import java.util.Set;
+
+import com.teragrep.blf_01.Tokenizer;
+
+import static com.teragrep.pth_06.jooq.generated.bloomdb.Bloomdb.BLOOMDB;
+
+
+public final class IndexStatementCondition implements QueryCondition {
+    private final Logger LOGGER = LoggerFactory.getLogger(IndexStatementCondition.class);
+
+    private final Condition condition;
+    private final Element element;
+    private final ConditionConfig config;
+    private final Tokenizer tokenizer;
+
+    public IndexStatementCondition(Condition condition, Element element, ConditionConfig config) {
+        this.condition = condition;
+        this.element = element;
+        this.config = config;
+        this.tokenizer = new Tokenizer(32);
```

Comment:
Should there be a primary constructor where the Tokenizer can be provided, and include this one as a secondary constructor?

Reply:
added a primary constructor